### PR TITLE
unix: fail fast when pkg-config not found

### PR DESCRIPTION
**Pull request by condemil**
Prevents many misleading messages during build when pkg-config is not installed.

### CODE REVIEW

File: ports/unix/Makefile [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Fail fast when pkg-config is not installed
+ifeq ($(shell which pkg-config || echo 0), 0)
```
Comment (dlech) [inline ports/unix/Makefile:2]: `pkg-config` is only required for specific configurations (not `MICROPY_STANDALONE`  or `MICROPY_BLUETOOTH_BTSTACK` so this check should probably be guarded by the same conditions.

File: ports/unix/Makefile [hunk 2]
```diff
@@ -1,3 +1,20 @@
+# Fail fast when pkg-config is not installed and required
+PKG_CONFIG_REQUIRED = 0
+
+ifeq ($(MICROPY_BLUETOOTH_BTSTACK),1)
+PKG_CONFIG_REQUIRED = 1
+endif
+
+ifeq ($(MICROPY_STANDALONE),1)
```
Comment (dlech) [inline ports/unix/Makefile:8]: `pkg-config` is used in the `else` of this check so it needs to be negated.

```suggestion
ifneq ($(MICROPY_STANDALONE),1)
```

File: ports/unix/Makefile [hunk 3]
```diff
@@ -1,3 +1,20 @@
+# Fail fast when pkg-config is not installed and required
+PKG_CONFIG_REQUIRED = 0
+
+ifeq ($(MICROPY_BLUETOOTH_BTSTACK),1)
```
[the remark this answers is not in the document]
Comment (dlech) [reply ports/unix/Makefile:4]: Although this case already has a `which pkg-config` test, so doesn't seem to actually be required in this case.

**Review by dlech — COMMENTED** [review]
After looking at this a bit harder, I see that `pkg-config` is optional for the Bluetooth case, so this might be simpler to just put the original check inside of the `else` clause where `pkg-config` is actually used.

**Comment by jimmo** [comment]
Thanks @condemil 

I think for now a simpler option might be to ignore the BLE use case (which already does something sensible enough if pkg-config isn't available), and just handle the direct requirement of detecting pkg-config not available for FFI.

The following would work:

```
LIBFFI_CFLAGS := $(shell pkg-config --cflags libffi)
LIBFFI_LDFLAGS := $(shell pkg-config --libs libffi)
ifneq ($(.SHELLSTATUS),0)
$(error pkg-config is required for MICROPY_PY_FFI.)
endif
```

**Pull request closed without merging**